File: Sample Codes/Delphi/Device Programming/Dialog.pas
unit Dialog;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  ACR122s, Dialogs, StdCtrls;

type
    TFDialog = class(TForm)
    Label1: TLabel;
    Label2: TLabel;
    Label3: TLabel;
    Label4: TLabel;
    tbStatTO: TEdit;
    tbStatRe: TEdit;
    tbRespTO: TEdit;
    tbRespRe: TEdit;
    bOk: TButton;
    bCancel: TButton;
    procedure bCancelClick(Sender: TObject);
    procedure FormShow(Sender: TObject);
    procedure bOkClick(Sender: TObject);
    procedure tbStatTOKeyPress(Sender: TObject; var Key: Char);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  FDialog: TFDialog;

implementation

uses DeviceProgramming;

{$R *.dfm}

procedure TFDialog.bCancelClick(Sender: TObject);
begin
    FDialog.Close;
end;

procedure TFDialog.bOkClick(Sender: TObject);
var
acrTO : ACR122_TIMEOUTS;
retCode : Integer;

begin

   if tbStatTo.Text = '' then begin
      tbStatTo.SetFocus;
      Exit;
   end;

   if tbStatRe.Text = '' then begin
      tbStatRe.SetFocus;
      Exit;
   end;

   if tbRespTo.Text = '' then begin
      tbRespTo.SetFocus;
      Exit;
   end;

   if tbRespRe.Text = '' then begin
      tbRespRe.SetFocus;
      Exit;
   end;

   acrTO.statusTimeOut := StrToInt(tbStatTo.Text);
   acrTO.numStatusRetries := StrToInt(tbStatRe.Text);
   acrTO.responseTimeout := StrToInt(tbRespTo.Text);
   acrTO.numResponseRetries := StrToInt(tbRespRe.Text);

   retCode := ACR122_SetTimeouts( DeviceProgramming.hReader, @acrTo);

   if retCode = 0 then begin
      displayOut( 0, 0, 'Set Timeouts success');
   end
   else begin
      displayOut( 1, 0, 'Set Timeouts failed');
   end;

end;

procedure TFDialog.FormShow(Sender: TObject);
begin
   tbStatTO.Text := '2000';
   tbStatRe.Text := '1';
   tbRespTo.Text := '10000';
   tbRespRe.Text := '1';
end;

procedure TFDialog.tbStatTOKeyPress(Sender: TObject; var Key: Char);
begin
  if Key <> chr($08) then
    if Not (Key in ['0'..'9'])then
      Key := Chr($00);
end;

end.

File: Sample Codes/Visual Basic .Net/DeviceProgramming/DeviceProgramming/Dialog.vb
Imports System.Windows.Forms

Public Class Dialog

    Private Sub OK_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Me.DialogResult = System.Windows.Forms.DialogResult.OK
        Me.Close()
    End Sub

    Private Sub Cancel_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Me.DialogResult = System.Windows.Forms.DialogResult.Cancel
        Me.Close()
    End Sub

    Private Sub Dialog_Shown(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Shown
        tbStatTO.Text = "2000"
        tbStatRe.Text = "1"
        tbRespTO.Text = "10000"
        tbRespRe.Text = "1"
    End Sub

    Private Sub bCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles bCancel.Click
        Me.Close()

    End Sub

    Private Sub tbStatTO_KeyPress(ByVal sender As System.Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles tbStatTO.KeyPress, tbStatRe.KeyPress, tbRespTO.KeyPress, tbRespRe.KeyPress
        If InStr("0123456789", e.KeyChar) > 0 Or e.KeyChar = Microsoft.VisualBasic.vbBack Then

        Else
            e.KeyChar = ""

        End If
    End Sub

    Private Sub bOk_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles bOk.Click
        Dim timeOut As ACR122_TIMEOUTS

        If tbStatTO.Text = "" Then
            tbStatTO.Focus()
            Exit Sub
        End If

        If tbStatRe.Text = "" Then
            tbStatRe.Focus()
            Exit Sub
        End If

        If tbRespTO.Text = "" Then
            tbRespTO.Focus()
            Exit Sub
        End If

        If tbRespRe.Text = "" Then
            tbRespRe.Focus()
            Exit Sub
        End If

        timeOut.statusTimeout = CInt(tbStatTO.Text)
        timeOut.numStatusRetries = CInt(tbStatRe.Text)
        timeOut.responseTimeout = CInt(tbRespTO.Text)
        timeOut.numResponseRetries = CInt(tbRespRe.Text)

        DeviceProgramming.retCode = ACR122_SetTimeouts(DeviceProgramming.hReader, timeOut)

        If DeviceProgramming.retCode = 0 Then
            DeviceProgramming.DisplayMsg(0, 0, "Set Timeouts success")
        Else
            DeviceProgramming.DisplayMsg(1, 0, "Set Timeouts failed")
        End If


    End Sub
End Class
